org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ProgressReporterManager.java

```diff
@@ -177,20 +198,23 @@ protected String formatMessage(String task) {
 
 		@Override
 		protected String formatMessage(String task) {
-			String message = this.taskName == null || this.taskName.length() == 0 ? "" : ("- " + this.taskName);
-			if (task != null && task.length() > 0) {
-				message = message + " - " + task;
+			String message = getMessage(task);
+			if (totalWork > 0 && !message.isEmpty()) {
+				message = SEPARATOR + message;
 			}
-			return String.format("%.0f%% Starting Java Language Server %s", ((double) progress / totalWork) * 100, message);
+			return String.format("%.0f%% Starting Java Language Server%s", ((double) progress / totalWork) * 100, message);
```

Comment:
space should not be removed

Reply:
getMessage adds a space. You can try ProgressReporterManagerTest.testStartupJobReporting()